From 5889511618c74cf4ba7bc97c1774af4fbbf9f7af Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 25 Jul 2022 11:56:10 +0300
Subject: [PATCH] [SW-86607] gaudi3: print full preboot load addr

Preboot load address is stored in flash as the fullchip 64b address.
Respectively it should be printed as 64b.

Change-Id: Ief8ae0922959ce3516b07e55f007f17392a1f25c
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 36e3455..f2c6abe 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -86,12 +86,11 @@
 
 void ppboot_cp_preboot(u64 loadaddr, u32 offset, u32 size, char *name)
 {
-	u32 laddr = (u32)loadaddr;
 	u64 dma_dest_addr;
 	enum dma_error dma_err;
 	enum dma_dest_type dma_type;
 
-	preboot_puts_hex("Preboot loadaddr: ", laddr, false);
+	preboot_puts_hex64("Preboot loadaddr: ", loadaddr, false);
 
 	dma_dest_addr = gaudi3_dma_addr_fixup(loadaddr);
 	dma_type = preboot_get_dma_dest_type(dma_dest_addr);
